packages/compass-settings/src/stores/settings.ts

```diff
@@ -173,21 +229,37 @@ export const changeFieldValue = <K extends keyof UserConfigurablePreferences>(
   };
 };
 
-export const openModal = (): SettingsThunkAction<Promise<void>> => {
+export const openModal = (
+  tab?: SettingsTabId
+): SettingsThunkAction<Promise<void>, OpenSettingsModalAction> => {
   return async (dispatch) => {
-    dispatch({ type: ActionTypes.OpenSettingsModal });
+    dispatch({ type: ActionTypes.OpenSettingsModal, tab });
     await dispatch(fetchSettings());
   };
 };
 
-export const closeModal = (): SettingsThunkAction<void> => {
+export const selectTab = (
+  tab?: SettingsTabId
+): SettingsThunkAction<void, SelectTabAction> => {
+  return (dispatch) => {
+    dispatch({ type: ActionTypes.SelectTab, tab });
+  };
+};
```

Comment:
Small nit: doesn't need to be a thunk action. Nothing wrong with it, just you usually might want to use those only if you need to dispatch multiple events from one action, access current state before dispatching, or have access to the thunk "extra arg". None of these are the case here

```suggestion
): SelectTabAction => {
  return { type: ActionTypes.SelectTab, tab }
};
```

Reply:
Sure – done in b3a9e6f02dacd0e33a2a6b8eb959bf1aca436534 👍